packages/compartment-mapper/src/import-hook.js

```diff
@@ -142,6 +150,18 @@ export const makeImportHookMaker = (
             record = concreteRecord;
           }
 
+          let sha512;
+          if (computeSha512 !== undefined) {
+            sha512 = computeSha512(transformedBytes);
+            if (expectedSha512 !== undefined && sha512 !== expectedSha512) {
```

Comment:
Why is the expected hash optional here, when the earlier check (in `makeImportHook`) makes it mandatory (both in the case where the ability to hash at all has been provided)?

Reply:
This is also a good catch. The expectedSha512 is in fact always undefined because this import hook maker only gets used when creating a compartment map from scratch, so we don’t need the code above to extract expectedSha512 (never exists) nor the code below to test it, just this bit in the middle to compute a new one if the power is provided.